action/protocol/execution/protocol.go

```diff
@@ -62,9 +63,36 @@ func FindProtocol(registry *protocol.Registry) *Protocol {
 
 // Handle handles an execution
 func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.StateManager) (*action.Receipt, error) {
+	featureCtx := protocol.MustGetFeatureCtx(ctx)
 	exec, ok := act.(*action.Execution)
 	if !ok {
-		return nil, nil
+		if !featureCtx.TransferToContractFallback {
+			return nil, nil
+		}
+
+		tsf, ok := act.(*action.Transfer)
+		if !ok {
```

Comment:
`if !ok || len(tsf.Payload()) == 0`
receive/fallback is for a call with empty calldata

Reply:
data could be nonempty

Reply:
from: https://docs.soliditylang.org/en/v0.6.0/contracts.html#receive-ether-function

This function cannot have arguments, cannot return anything and must have external visibility and payable state mutability. It is executed on a call to the contract with **empty calldata**.

Reply:
if allow non-empty data, it could end up calling an existing valid `payable` function

for example, the contract has a function `receiveFromFool() payable`, with func signature = `0x1234abcd`

then a transfer with payload = `0x1234abcd` will be converted to a call to `receiveFromFool()`, we need to prevent that

Reply:
or do we want to allow such interpretation/convertion?